### DISCUSSION TITLE
How can I increase the shadow map resolution?

### DISCUSSION THREAD

**Discussion by harrycollin** [heremaps/harp.gl#2224] [Q&A] [answered]
I'd like to increase the resolution of the shadows. They seem to have defaulted to 512x512 for directional lights. When I try to update this value the shadows move all over the tiles like crazy.

**Comment by nzjony** [comment, marked as answer]
Hi @harrycollin , you can change it here: https://github.com/heremaps/harp.gl/blob/81f8e593c683250acfd9c206d0bc6cc4ddc1489b/%40here/harp-mapview/lib/ThemeHelpers.ts#L133

I changed to 2048x2048 and it worked for me.

**Reply by nzjony** [reply]
You can also change it at runtime, if you do something like:

(assuming the light is called `sun`)
```
    (sun.shadow.map as any).dispose();
    sun.shadow.mapSize.width = 2048;
    sun.shadow.mapSize.height = 2048;
```

**Comment by harrycollin** [comment]
When I do this the shadows don't seem to sync up with the map. When I move the camera the shadows slide around all over the tiles. 
![image](https://user-images.githubusercontent.com/10602138/123491107-d87e3200-d60d-11eb-8943-593d11245f57.png)
Any ideas what could cause this?

I'm following this method you proposed.
![image](https://user-images.githubusercontent.com/10602138/123491149-ef248900-d60d-11eb-8e67-00d3c5de30b6.png)

**Reply by nzjony** [reply]
@harrycollin , did you try changing in ThemeHelpers.ts?

**Reply by harrycollin** [reply]
I haven't, I was hoping this could be configured without modifying the src code. I'll try it there and see what happens.

**Reply by nzjony** [reply]
@harrycollin , did you manage to solve the issue?

Could you change it at run time without changing the harp source code? If not, I can try to look into it again.

**Reply by nzjony** [reply]
@harrycollin I managed to figure it out, you also need to add: (sun.shadow as any).map = null;

Sorry about that! I will update the shadow demo to show how to change it at runtime.

**Reply by nzjony** [reply]
@harrycollin  see: https://github.com/heremaps/harp.gl/pull/2239

**Reply by harrycollin** [reply]
@nzjony that worked! Thanks :)
